File: app/api/src/ai_interviewer_api/core/interview_configuration.py
from collections.abc import Mapping

from fastapi import HTTPException

from ai_interviewer_api.models.interview_plan import STRUCTURED_INTERVIEW_MODEL_IDS


INTERVIEW_PROFILES = frozenset({"fixed_form", "business_process", "system_requirement"})


def is_interview_configuration_complete(knowledge: Mapping[str, object]) -> bool:
    """Return whether the saved knowledge can create and start an interview record."""

    plan = knowledge.get("interviewPlan")
    if not isinstance(plan, Mapping):
        return False
    return (
        plan.get("profile") in INTERVIEW_PROFILES
        and plan.get("modelId") in STRUCTURED_INTERVIEW_MODEL_IDS
    )


def require_interview_configuration(knowledge: Mapping[str, object]) -> None:
    if not is_interview_configuration_complete(knowledge):
        raise HTTPException(status_code=409, detail="interview_configuration_required")

File: app/web/src/features/interviews/interviewConfiguration.ts
import type { Knowledge } from "@ai-interviewer/shared-types";

export const DEFAULT_INTERVIEW_MODEL_ID = "global.openai.gpt-5.6-luna" as const;

export function isInterviewConfigurationComplete(knowledge: Knowledge | null) {
  const plan = knowledge?.interviewPlan;
  return Boolean(
    plan
      && ["fixed_form", "business_process", "system_requirement"].includes(plan.profile ?? "")
      && ["global.openai.gpt-5.6-terra", "global.openai.gpt-5.6-luna"].includes(plan.modelId ?? ""),
  );
}
